lib/fluent/plugin/service_discovery/discovery_message.rb

```diff
@@ -0,0 +1,38 @@
+#
+# Fluentd
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+require 'fluent/plugin/service_discovery'
+
+module Fluent
+  module Plugin
+    class ServiceDiscovery # already loaded
+      DiscoveryMessage = Struct.new(:type, :service) do
```

Comment:
Exposing this struct is needed? `ServiceDiscovery.service_in/service_out` also looks good for me.
With this approach, we can move this implementation into `service_discovery.rb` and users don't need to require additional file.

Reply:
fixed https://github.com/fluent/fluentd/pull/2541/commits/4ba7c7a40e8184f6d3f685482a9f24931cd87277